chainercv/datasets/cityscapes/cityscapes_semantic_segmentation_dataset.py

```diff
@@ -0,0 +1,85 @@
+import glob
+import os
+
+import numpy as np
+
+from chainer import dataset
+from chainercv.datasets.cityscapes.cityscapes_utils import cityscapes_labels
+from chainercv.utils import read_image
+
+
+class CityscapesSemanticSegmentationDataset(dataset.DatasetMixin):
+
+    """Dataset class for a semantic segmentation task on `Cityscapes dataset`_.
+
+    .. _`Cityscapes dataset`: https://www.cityscapes-dataset.com
+
+    .. note::
+
+        Please download the data by yourself because Cityscapes dataset doesn't
+        allow to re-distribute their data.
+
+    Args:
+        img_dir (string): Path to the image dir. It should end with
+            ``leftImg8bit``.
+        label_dir (string): Path to the dir which contains labels. It should
+            end with either ``gtFine`` or ``gtCoarse``.
+        split ({'train', 'val'}): Select from dataset splits used in
+            Cityscapes dataset.
+        ignore_labels (bool): If True, the labels marked ``ignoreInEval``
+            defined in the original
+            `cityscapesScripts<https://github.com/mcordts/cityscapesScripts>_`
+            will be replaced with `-1` in the `get_example` method.
+
+    """
+
+    def __init__(self, img_dir, label_dir, split='train', ignore_labels=True):
+        img_dir = os.path.join(img_dir, split)
+        self.ignore_labels = ignore_labels
+
+        self.label_fns, self.img_fns = [], []
+        resol = os.path.basename(label_dir)
+        for dname in glob.glob('{}/*'.format(label_dir)):
+            if split in dname:
+                for label_fn in glob.glob(
+                        '{}/*/*_labelIds.png'.format(dname)):
+                    self.label_fns.append(label_fn)
+        for label_fn in self.label_fns:
```

Comment:
`*_paths` is shorter than `*_filenames` and `_fnames`. If the problem of `*_filenames` is its length, `*_paths` will be a good solution.